File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        head=head.next
        if not head:
            return
        temp=head
        res=0
        while temp.val!=0:
            res+=temp.val
            temp=temp.next
        head.val=res
        head.next=self.mergeNodes(temp)
        return head
        
        

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        head=head->next;
        if(!head){
            return NULL;
        }
        ListNode* temp=head;
        int res=0;
        while(temp->val!=0){
            res+=temp->val;
            temp=temp->next;
        }
        head->val=res;
        head->next=mergeNodes(temp);
        return head;
    }
};